src/Core/Layer/Collector/PackageNameCollector.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Qossmic\Deptrac\Core\Layer\Collector;
+
+use PHPStan\PhpDocParser\Ast\PhpDoc\PhpDocTagNode;
+use PHPStan\PhpDocParser\Lexer\Lexer;
+use PHPStan\PhpDocParser\Parser\ConstExprParser;
+use PHPStan\PhpDocParser\Parser\PhpDocParser;
+use PHPStan\PhpDocParser\Parser\TokenIterator;
+use PHPStan\PhpDocParser\Parser\TypeParser;
+use Qossmic\Deptrac\Contract\Ast\CouldNotParseFileException;
+use Qossmic\Deptrac\Contract\Ast\TokenReferenceInterface;
+use Qossmic\Deptrac\Contract\Layer\InvalidCollectorDefinitionException;
+use Qossmic\Deptrac\Core\Ast\AstMap\ClassLike\ClassLikeReference;
+use Qossmic\Deptrac\Core\Ast\Parser\NikicPhpParser\NikicPhpParser;
+
+class PackageNameCollector extends RegexCollector
+{
+    private readonly Lexer $lexer;
+    private readonly PhpDocParser $docParser;
+
+    public function __construct(
+        private readonly NikicPhpParser $nikicPhpParser
+    ) {
+        $this->lexer = new Lexer();
```

Comment:
I agree. The collector should not be aware of the parser. I think this should be an extractor instead, which can then emit a reference. I am not sure, if it should be a class reference or maybe something else/new.

Reply:
At the moment the TokenReferenceInterface only handles the Token and its filepath - could we add a more generic array of meta information, which we can then search later in collectors?

For example, if there was a new TokenReferenceMetaInterface, with an array of those in each TokenReferenceInterface, we could store a new PackageAnnotationMeta object in that array when we find the `@package` annotation, and then search for it in collectors.

This would prevent the TokenReferenceInterface gradually having more fields as more meta data is attached to it (if more collectors come along in the future).

The extractor you suggest could then add a TokenReferenceMeta to the TokenReference.

Sorry - I don't think I've explained this very well - I'll see if I can start to build it so you can see what I mean.